root/edit/details/EditEvent.js

```diff
@@ -0,0 +1,114 @@
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import FullChangeDiff from
+  '../../static/scripts/edit/components/edit/FullChangeDiff';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+
+type EditEventProps = {
+  +display_data: {
+    +begin_date?: CompT<PartialDateT | null>,
+    +cancelled?: CompT<boolean>,
+    +comment?: CompT<string | null>,
+    +end_date?: CompT<PartialDateT | null>,
+    +event: EventT,
+    +name?: CompT<string>,
+    +setlist?: CompT<string | null>,
+    +time?: CompT<string | null>,
+    +type?: CompT<EventT | null>,
+  },
+};
+
+const EditEvent = ({edit}: {+edit: EditEventProps}) => {
+  const display = edit.display_data;
+  const name = display.name;
+  const comment = display.comment;
+  const cancelled = display.cancelled;
+  const type = display.type;
+  const beginDate = display.begin_date;
+  const endDate = display.end_date;
+  const time = display.time;
+  const setlist = display.setlist;
+  return (
+    <table className="details edit-event">
+      <tr>
+        <th>{addColonText(l('Event'))}</th>
+        <td colSpan="2"><EntityLink entity={display.event} /></td>
+      </tr>
+      {name ? (
+        <Diff
+          label={addColonText(l('Name'))}
+          newText={name.new}
+          oldText={name.old}
+          split="\s+"
+        />
+      ) : null}
+      {comment ? (
+        <Diff
+          label={addColonText(l('Disambiguation'))}
+          newText={comment.new ?? ''}
+          oldText={comment.old ?? ''}
+          split="\s+"
+        />
+      ) : null}
+      {cancelled ? (
+        <FullChangeDiff
+          label={addColonText(l('Cancelled'))}
+          newContent={yesNo(cancelled.new)}
+          oldContent={yesNo(cancelled.old)}
+        />
+      ) : null}
+      {type ? (
+        <FullChangeDiff
+          label={addColonText(l('Type'))}
+          newContent={type.new?.name ?? ''}
+          oldContent={type.old?.name ?? ''}
+        />
+      ) : null}
+      {beginDate ? (
+        <Diff
+          label={addColonText(l('Begin date'))}
+          newText={formatDate(beginDate.new)}
+          oldText={formatDate(beginDate.old)}
+          split="-"
+        />
+      ) : null}
+      {endDate ? (
+        <Diff
+          label={addColonText(l('End date'))}
+          newText={formatDate(endDate.new)}
+          oldText={formatDate(endDate.old)}
+          split="-"
+        />
+      ) : null}
+      {time ? (
+        <FullChangeDiff
+          label={addColonText(l('Time'))}
+          newContent={time.new}
+          oldContent={time.old}
```

Comment:
I see in the typing that time can be null. If that's true, then we should be checking it here, no? And if it's not, we should change the typing. Unless I'm missing something.

Reply:
I see now that the diff accepts react.node, so it *is* allowed to be null - but then it'd make sense for all others to also just pass null when it's null, no?

Reply:
As for this.. I've added `?? ''` to check if the value is null and pass an empty string instead. I think that should better.